arrows/core/applets/dump_klv.cxx

```diff
@@ -44,7 +50,9 @@ add_command_options()
     ( "h,help", "Display applet usage" )
     ( "c,config", "Configuration file for tool", cxxopts::value<std::string>() )
     ( "o,output", "Dump configuration to file and exit", cxxopts::value<std::string>() )
+    ( "l,log", "Log metadata to a .json file.", cxxopts::value<std::string>() )
```

Comment:
This specifically indicates that the output will be JSON, but it is possible to specify a different type of output using the config file. If you are only supporting JSON, you could programmatically force that setting into the config. this would also get around a possible error loading the serializer if it is not selected on the command line and not available in the build (as described below).

Also, the JSON output is in a specific format that is not exactly the same as the stdout dump. Maybe all that is required is to explain the specific metadata map format. If somebody else wants to use JSON format, it should be clear what they are getting.

Reply:
I added a rather lengthy explanation here. I'm happy to pare it down if this is too much.